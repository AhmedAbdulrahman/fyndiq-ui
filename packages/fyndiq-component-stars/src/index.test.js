import React from 'react'
import renderer from 'react-test-renderer'

import Stars from './'

describe('fyndiq-component-stars', () => {
  beforeAll(() => {
    Math.random = jest.fn(() => 0.5)
  })

  it('should be rendered without props', () => {
    const component = renderer.create(<Stars />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should be rendered with prop rating', () => {
    const component = renderer.create(<Stars rating={4.5} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should be rendered with prop reviews', () => {
    const component = renderer.create(<Stars reviews={4} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should be rendered with prop size', () => {
    const component = renderer.create(<Stars size="l" />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should be rendered with prop reviews', () => {
    const component = renderer.create(<Stars interactive />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
