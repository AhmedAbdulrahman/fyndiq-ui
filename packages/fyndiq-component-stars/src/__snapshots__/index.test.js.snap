// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-stars should be rendered with prop interactive 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              interactive
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should be rendered with prop rating 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0.5}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should be rendered with prop reviews 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
  <span
    className="reviews"
  >
    (
    4
     reviews)
  </span>
</div>
`;

exports[`fyndiq-component-stars should be rendered with prop size 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-l
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should be rendered without props 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should change its state when hovered and interactive 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              interactive
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should not change its state when hovered and not interactive 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should reset its state when the mouse leaves 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              interactive
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      key="1"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="2"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="3"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      key="4"
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0.2999999999999998}
      key="5"
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;
